### DISCUSSION TITLE
Set lock entity code validation by default as feature or attribute

### DISCUSSION THREAD

**Discussion by jbouwh** [home-assistant/architecture#872] [Entity Models]
The Lock entity now validates a code that is passed to a service call when code_format is set.
In cases where the absence of a code should be supported https://github.com/home-assistant/architecture/discussions/870 (https://github.com/home-assistant/core/pull/87544 suggests) we should make a bypass possible (now it is mandatory if `code_format` is set).

With this change implemented the proposal is to make code validation default using an extra LockEntityFeature (Example: https://github.com/home-assistant/core/pull/87588). (See also: https://github.com/home-assistant/core/pull/87544#discussion_r1097981448)
Alternative could be adding an attribute enable code validation by default, e.g. _attr_code_default.

**Comment by balloob** [comment]
If code is not mandatory, why even accept a code at all?

**Reply by jbouwh** [reply]
A code is now mandatory when code_format is set. The suggestion done with #87544 would lead to only validate a code when it is given. May be the conclusion is that [#87544](https://github.com/home-assistant/core/pull/87544) is a not wanted change?

**Reply by balloob** [reply]
But why would people even set a code format if we don't need a code?

**Reply by jbouwh** [reply]
May be a code is not needed for opening?
